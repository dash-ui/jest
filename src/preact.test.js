/** @jsx h */
// eslint-disable-next-line
import {h} from 'preact'
import render from 'preact-render-to-json'
import prettyFormat from 'pretty-format'
import {createStyle} from 'test-utils'
import serializer, {createSerializer} from 'index'
const {ReactElement, ReactTestComponent, DOMElement} = prettyFormat.plugins

expect.addSnapshotSerializer(serializer)
let dashPlugin = createSerializer()
afterEach(() => {
  document.getElementsByTagName('html')[0].innerHTML = ''
})

describe('@-ui/jest with preact', () => {
  it('replaces class names and inserts styles into preact test component snapshots', () => {
    const style = createStyle()
    const tree = render(
      <div class={style('div')}>
        <svg class={style('svg')} />
      </div>
    )

    const output = prettyFormat(tree, {
      plugins: [dashPlugin, ReactElement, ReactTestComponent, DOMElement],
    })

    expect(output).toMatchSnapshot()
  })

  it('handles elements with no props', () => {
    const tree = render(<div />)

    const output = prettyFormat(tree, {
      plugins: [dashPlugin, ReactElement, ReactTestComponent, DOMElement],
    })

    expect(output).toMatchSnapshot()
  })
})
