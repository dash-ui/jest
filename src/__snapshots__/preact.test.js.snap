// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@-ui/jest with preact handles elements with no props 1`] = `"<div />"`;

exports[`@-ui/jest with preact replaces class names and inserts styles into preact test component snapshots 1`] = `
".-ui-1 {
  color: red;
}

.-ui-0 {
  width: 100%;
}

<div
  children={
    <svg
      class=\\"-ui-0\\"
    />
  }
  class=\\"-ui-1\\"
/>"
`;
